### PR TITLE
Sampling: poc for 1p and 3p sampling 

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public override void OnStartActivity(Activity activity, object payload)
             activity.DisplayName = path;
 
             activity.SetKind(ActivityKind.Server);
-
+            activity.SetCustomProperty("OTel.HttpInHandler.Request", request);
```
Comment (Kanwaldeep) [inline src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs]: Do we need this change? I don't think we added this for the other example Amrit worked on?

File: src/OpenTelemetry/Trace/ActivitySourceAdapter.cs [hunk 2]
```diff
@@ -107,6 +107,10 @@ private void RunGetRequestedData(Activity activity)
                 activity.Links);
 
             var samplingResult = this.sampler.ShouldSample(samplingParameters);
+            foreach (var attr in samplingResult.Attributes)
```
Comment (Kanwaldeep) [inline src/OpenTelemetry/Trace/ActivitySourceAdapter.cs]: Is this something OTel is planning to do or we will need to create an issue

File: OpenTelemetry.Azure.Sampling/ThirdPartyParentOnlySampler.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            // if it's server span, and it's on the public endpoint
+            if ((this.boundaries & PublicBoundary.Incoming) == PublicBoundary.Incoming &&
+                this.IsServerSpan(in samplingParameters))
+            {
+                string resourceId = this.getResourceIdCallback?.Invoke();
+
+                // follow parent decision + respect sampling configuration settings
+                if (!string.IsNullOrEmpty(resourceId) && 
+                    samplingParameters.ParentContext.TraceFlags == ActivityTraceFlags.Recorded &&
+                    this.samplingCallback(resourceId))             // samplingCallback(resoureId, GetScore(traceId)) <= this.rate for rate-based
```
Comment (Kanwaldeep) [inline OpenTelemetry.Azure.Sampling/ThirdPartyParentOnlySampler.cs:50]: So for rate based is the idea to have a GetScore method that would calculate the score and then the callback could make a decision.

Comment (lmolkova) [reply OpenTelemetry.Azure.Sampling/ThirdPartyParentOnlySampler.cs:50]: yes, getscore will be an internal rate-based sampler implementation detail

File: src/OpenTelemetry/Trace/ActivitySourceAdapter.cs (hunk 2 above)
Comment (lmolkova) [reply src/OpenTelemetry/Trace/ActivitySourceAdapter.cs]: it's being addressed in otel

File: src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs (hunk 1 above)
Comment (lmolkova) [reply src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs]: this is an accident

**Pull request closed without merging**